Support pre-hashed passwords with userpass backend (#25862)
hashicorp/vault: website/content/docs/auth/userpass.mdx

1. Enable the userpass auth method:

  ```shell-session
  $ vault auth enable userpass
  ```

>   This enables the userpass auth method at `auth/userpass`. To enable it at a different path, use the `-path` flag:

Reviewer: Suggest simplifying the instructions by adding this to the text before the example command as:

Enable the userpass auth method at the default `auth/userpass` path. You can choose to enable the auth method at a different path with the `-path` flag:

  ```shell-session
  $ vault auth enable -path=<path> userpass
  ```

1. Configure it with users that are allowed to authenticate:

  ```shell-session
  $ vault write auth/<userpass:path>/users/mitchellh \
      password=foo \
      policies=admins
  ```

>   This creates a new user "mitchellh" with the password "foo" that will be
>   associated with the "admins" policy. This is the only configuration
>   necessary.

Reviewer: ```suggestion

The result is a new user, "mitchellh", with the password "foo", who is 
associated with the "admins" policy. 
```

Style correction: avoid "this" as a pronoun.
